otherlibs/ocamlc_loc/src/lexer.mll

```diff
@@ -0,0 +1,127 @@
+{
+  (* raise when the format is unrecognized *)
+  exception Unknown_format
+
+  type lines =
+    | Single of int
+    | Range of int * int
+
+  type source =
+    | Code of { code : int ; name : string }
+    | Alert of string
+
+  type severity =
+    | Error of source option
+    | Warning of source
+
+  type loc =
+    { chars : (int * int) option
+    ; lines : lines
+    ; path : string
+    }
+
+  type line = { indent : int ; contents : string }
+
+  type token =
+    | Toplevel of
+      { indent : int
+      ; loc : loc
+      ; severity : severity
+      ; message : string
+      }
+    | Related of { indent : int ; loc : loc ; message: string }
+    | Line of line
+    | Eof
+
+  let parse_range s =
+    match String.split_on_char '-' s with
+    | [x; y] -> (int_of_string x, int_of_string y)
+    | _ -> assert false
+}
+
+let blank = [' ' '\t' '\r']*
+
+let newline = '\n'
+
+let quoted_string = '"' [^ '"']+ '"'
+
+let digits = ['0' - '9']+
+
+let range = digits "-" digits
+
+rule skip_excerpt = parse
+  | digits " | " [^ '\n']* "\n"
+    { skip_excerpt lexbuf }
+  | blank '^'+ blank "\n"
+    { () }
+  | eof { () }
+  | "" { () }
+
+and severity = parse
+  | "Error:" blank { Error None }
+  | "Warning" blank (digits as code) blank "[" ([^ ']']+ as name) "]:" blank
+    { Warning (Code { code = int_of_string code ; name })
+    }
+  | "Error" blank
+    "(warning" blank (digits as code) blank "[" ([^ ']']+ as name) "]):"
+    blank
+    { Error (Some (Code { code = int_of_string code ; name }))
+    }
+  | (("Error" | "Warning") as kind) " (alert " ([^ ')']+ as alert) "):"
+    { let alert = Alert alert in
+      match kind with
+      | "Error" -> Error (Some alert)
+      | "Warning" -> Warning alert
+      | _ -> assert false
+    }
+  | "" { raise_notrace Unknown_format }
+
+and line = parse
+  | (blank as prefix) ([^ '\n']* as contents) blank newline?
+    { Line { indent = String.length prefix ; contents }
+    }
+  | eof { Eof }
+
+and toplevel_message = parse
+  | blank ([^ '\n']* as message) blank '\n' { message }
+  | "" { "" }
+
+and token = parse
+  | (blank as indent) "File \"" ([^ '"']* as path) "\", " blank
+    (("line " (digits as line) | "lines " (range as lines)))
+    ("," blank "characters" blank (range as chars))?
```

Comment:
I don't know if OCaml does this, but if there is a 1 char range, does it default to `character`?

Reply:
I checked the ocaml source and `character` is indeed impossible.